include/aspect/time_stepping/convection_time_step.h

```diff
@@ -0,0 +1,62 @@
+/*
+  Copyright (C) 2014 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_time_stepping_convection_time_step_h
+#define _aspect_time_stepping_convection_time_step_h
+
+#include <aspect/time_stepping/interface.h>
+
+
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    using namespace dealii;
+
+    /**
+     *
+     *
+     * @ingroup TimeStepping
+     */
+    template <int dim>
+    class ConvectionTimeStep : public Interface<dim>, public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Constructor.
+         */
+        ConvectionTimeStep () = default;
+
+
+        /**
+         * The main execute() function.
```

Comment:
Personally I would find it useful to have here similair (or the same) information which you have in the interface class for this function.